### ISSUE TITLE
embed/object

### ISSUE THREAD

**Issue by annevk** [annevk/orb#5]
It seems to me that the current way https://fetch.spec.whatwg.org/#corb is described it does not work well with the `embed` and `object` elements, as something gets blocked before those elements get to decide to instantiate a browsing context and both elements can render HTML.

I suspect this is an oversight in the formal model as bypassing the CORB check for these elements should be safe given that nested browsing contexts ought to get their own process when appropriate.

cc @anforowicz

**Comment by anforowicz** [comment]
I see.  I think you are hitting a problem similar to https://crbug.com/929300.  I tried to re-read the discussion in that bug, but I am not sure if I understood much from it.

1) NetworkService in Chromium enforces that only the supervisor/browser process may trigger `navigate` requests.  Renderer processes may ask the browser process to start a navigation, but the request (and response data) is controlled by the browser process.
2) CORB in Chromium applies to all requests from a renderer hosting http content (including <object src="https://other.origin.com/x.pdf"> in a https://example.com/page.html).
3) CORB in Chromium does not apply to requests from extensions.  PDFs in Chromium are renderered from an extension:
        example.com/page.html  --embeds-->  artificially conjured and generally invisible to web content chrome-extension:// iframe   --embeds--> PDF
4) Somehow (magic to me... :-( ) Chromium supports cross-origin PDFs in `<object>`, despite the restrictions above.  I think the answer lies within https://crbug.com/929300, but I am not sure how exactly it all works together...

**Comment by anforowicz** [comment]
> I suspect this is an oversight in the formal model as bypassing the CORB check for these elements should be safe given that nested browsing contexts ought to get their own process when appropriate.

It is indeed weird that some `<object>` elements (e.g. ones hosting Flash) can use `no-cors` fetches, but some (e.g. ones hosting PDF) use `navigate`.  Maybe with the deprecation of PPAPI, we could eventually say that all '<object>' elements trigger a `navigate` request?

**Comment by annevk** [comment]
Yeah, this is about the theoretical model. I do want to investigate if we can move to using navigate exclusively for `embed` and `object` elements, but I suspect that is not possible due to images.

The way `embed`/`object` roughly (there are a ton of issues with the current specification) work theoretically is that initially they do a no-cors request and then inspect the response and then either render the response (if it's an image) or create a nested browsing context and navigate to the response (e.g., if it's `text/html`). In implementations that logic will be split across process boundaries.

What Chrome seems to do for PDFs matches what it would do for `text/html` and I think matches what Firefox does as well. Create a nested browsing context and navigate it to the PDF. I think that means that only images can end up in the problematic process at which point it's fine to bypass CORB/ORB as long as the set of images that can up in the problematic process is not a larger set than CORB/ORB allows.

**Comment by annevk** [comment]
I just checked the HTML Standard and it already uses mode "`navigate`" for these requests (despite them not really being navigations). That will cause them to bypass this filter so we don't have to deal with them then.
